[BUG] missing `# tag: numpy` in nogil_buffer_acquisition test

### Describe the bug

Over on the weekly binary PyPy testing, I am seeing [an error](https://github.com/pypy/binary-testing/actions/runs/7771034626/job/21191877405#step:7:17577) when running `nogil_buffer_acquisition`:
```
Errors from shard 0:
======================================================================
FAIL: runTest (__mp_main__.CythonCompileTestCase)
[0] compiling (c/cy2) nogil_buffer_acquisition
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\binary-testing\binary-testing\cy\runtests.py", line 1106, in runTest
    self.runCompileTest()
  File "D:\a\binary-testing\binary-testing\cy\runtests.py", line 1110, in runCompileTest
    return self.compile(
  File "D:\a\binary-testing\binary-testing\cy\runtests.py", line 1413, in compile
    self._match_output(expected_errors, errors, tostderr)
  File "D:\a\binary-testing\binary-testing\cy\runtests.py", line 1461, in _match_output
    self.assertEqual(expected, actual)
AssertionError: '5:15: Buffer may not be acquired without[47 chars]ead.' != "3:8: 'numpy.pxd' not found"
- 5:15: Buffer may not be acquired without the GIL. Consider using memoryview slices instead.
+ 3:8: 'numpy.pxd' not found
``` 

The test is missing a `#tag: numpy`: https://github.com/cython/cython/blob/c271bc7b1f86390447834c4063a6f618b5dca51e/tests/errors/nogil_buffer_acquisition.pyx#L1-L10

It was last changed over 10 years ago, so I wonder why I am seeing the failure only now. Surely someone must have run the test suite without NumPy before?

### Code to reproduce the behaviour:

```cython
# example code
```
python runtest.py nogil_buffer_acquistion

### Expected behaviour

I would expect the test to be skipped if NumPy is not installed.

### OS

All

### Python version

All

### Cython version

HEAD

### Additional context

_No response_

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -176,12 +176,12 @@ jobs:
 
     steps:
       - name: Checkout repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
         with:
           fetch-depth: 1
 
       - name: Setup python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5.0.0
         with:
           python-version: ${{ matrix.python-version }}
 
@@ -198,14 +198,14 @@ jobs:
         run: bash ./Tools/ci-run.sh
 
       - name: Upload HTML docs
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4.3.0
         with:
           name: htmldocs
           path: docs/build/html
           if-no-files-found: ignore
 
       - name: Upload wheels
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4.3.0
         with:
           name: wheels-${{ runner.os }}${{ matrix.extra_hash }}
           path: dist/*.whl
@@ -222,10 +222,10 @@ jobs:
 
     steps:
       - name: Checkout repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
 
       - name: Setup python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5.0.0
         with:
           python-version: "3.11"
 
@@ -234,7 +234,7 @@ jobs:
         run: bash ./Tools/ci-run.sh
 
       - name: Upload Coverage Report
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4.3.0
         with:
           name: pycoverage_html
           path: coverage-report-html
@@ -249,12 +249,12 @@ jobs:
 
     steps:
       - name: Checkout repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
         with:
           fetch-depth: 1
 
       - name: Setup python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5.0.0
         with:
           python-version: "3.11"
 
@@ -263,7 +263,7 @@ jobs:
         run: bash ./Tools/ci-run.sh
 
       - name: Upload Coverage Report
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4.3.0
         with:
           name: cycoverage_html
           path: coverage-report-html
@@ -274,6 +274,6 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
       - name: Codespell
         uses: codespell-project/actions-codespell@v2

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -186,7 +186,7 @@ jobs:
           python-version: ${{ matrix.python-version }}
 
       - name: Compilation Cache
-        uses: hendrikmuhs/ccache-action@v1.2.9
+        uses: hendrikmuhs/ccache-action@v1.2.12
         with:
           variant: ${{ startsWith(runner.os, 'windows') && 'sccache' || 'ccache' }}  # fake ternary
           key: ${{ runner.os }}-hendrikmuhs-ccache${{ matrix.extra_hash }}-${{ matrix.python-version }}-${{ matrix.backend == 'c' || matrix.backend == 'c,cpp' }}-${{ contains(matrix.backend, 'cpp') }}-${{ hashFiles('test-requirements*.txt', '.github/**/ci.yml', 'Tools/**/ci-run.sh') }}
@@ -207,7 +207,7 @@ jobs:
       - name: Upload wheels
         uses: actions/upload-artifact@v4.3.0
         with:
-          name: wheels-${{ runner.os }}${{ matrix.extra_hash }}
+          name: wheels-${{ runner.os }}-${{ matrix.python-version }}${{ matrix.extra_hash }}
           path: dist/*.whl
           if-no-files-found: ignore
 

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -101,7 +101,7 @@ jobs:
 
     steps:
       - name: Checkout Cython
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
         
       - name: Set up QEMU
         if: runner.os == 'Linux' && !contains(matrix.only, 'x86') && !contains(matrix.only, 'i686')
@@ -127,7 +127,7 @@ jobs:
             ${{ contains(github.ref_name, 'a') || contains(github.ref_name, 'b')
               || contains(github.ref_name, 'rc') || contains(github.ref_name, 'dev') }}
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4.3.0
         with:
           name: ${{ matrix.only }}
           path: ./wheelhouse/*.whl
@@ -144,10 +144,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Cython
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4.1.1
 
       # Used to push the built wheels
-      - uses: actions/setup-python@v3
+      - uses: actions/setup-python@v5.0.0
         with:
           # Build sdist on lowest supported Python
           python-version: '3.8'
@@ -158,12 +158,12 @@ jobs:
           python setup.py sdist
           python setup.py bdist_wheel --no-cython-compile
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4.3.0
         with:
           name: sdist
           path: ./dist/*.tar.gz
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4.3.0
         with:
           name: pure-wheel
           path: ./dist/*.whl

--- a/tests/errors/nogil_buffer_acquisition.pyx
+++ b/tests/errors/nogil_buffer_acquisition.pyx
@@ -1,4 +1,5 @@
 # mode: error
+#tag: numpy
 
 cimport numpy as np
 

--- a/tests/errors/nogil_buffer_acquisition.pyx
+++ b/tests/errors/nogil_buffer_acquisition.pyx
@@ -7,5 +7,5 @@ cdef void func(np.ndarray[np.double_t, ndim=1] myarray) nogil:
     pass
 
 _ERRORS = u"""
-5:15: Buffer may not be acquired without the GIL. Consider using memoryview slices instead.
+6:15: Buffer may not be acquired without the GIL. Consider using memoryview slices instead.
 """

